### PR TITLE
CES Modules

### PR DESCRIPTION
**Pull request by SebastianGode**

### CODE REVIEW

File: plugins/modules/ces_alarms_info.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        if self.params['name']:
+            alarm = self.conn.ces.find_alarm(self.params['name'])
+            self.exit(
+                changed=False,
+                alarms=alarm
+            )
+
+        for raw in self.conn.ces.alarms():
+            dt = raw.to_dict()
```
Comment (anton-sidelnikov) [inline plugins/modules/ces_alarms_info.py:107]: here raw can be null

**Comment by tischrei** [comment]
recheck

File: plugins/modules/ces_alarms_info.py (hunk 1 above)
Comment (SebastianGode) [reply plugins/modules/ces_alarms_info.py:105]: Would it make a difference if it is null? Just to understand that

**Comment by SebastianGode** [comment]
I found no way to implement that as a module: https://docs.otc.t-systems.com/api/ces/en-us_topic_0032831274.html
The API somehow wants a List and even with the given Example I not managed to do it.

File: plugins/modules/ces_quotas_info.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = '''
+module: ces_quotas_info
+short_description: Get ressource Quotas
+extends_documentation_fragment: opentelekomcloud.cloud.otc
+version_added: "0.2.1"
```
Comment (tischrei) [inline plugins/modules/ces_quotas_info.py:18]: version is wrong

File: plugins/modules/ces_metrics_info.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = '''
+module: ces_metrics_info
+short_description: Get Metrics
+extends_documentation_fragment: opentelekomcloud.cloud.otc
+version_added: "0.2.1"
```
Comment (tischrei) [inline plugins/modules/ces_metrics_info.py:18]: version is wrong

File: plugins/modules/ces_metric_data_info.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = '''
+module: ces_metric_data_info
+short_description: Get Metric Data
+extends_documentation_fragment: opentelekomcloud.cloud.otc
+version_added: "0.2.1"
```
Comment (tischrei) [inline plugins/modules/ces_metric_data_info.py:18]: version is wrong

File: plugins/modules/ces_event_data_info.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = '''
+module: ces_event_data_info
+short_description: Get Event Data
+extends_documentation_fragment: opentelekomcloud.cloud.otc
+version_added: "0.2.1"
```
Comment (tischrei) [inline plugins/modules/ces_event_data_info.py:18]: version is wrong

File: plugins/modules/ces_alarms_info.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = '''
+module: ces_alarms_info
+short_description: Get Alarms
+extends_documentation_fragment: opentelekomcloud.cloud.otc
+version_added: "0.1.2"
```
Comment (tischrei) [inline plugins/modules/ces_alarms_info.py:18]: version is wrong

**Review by tischrei — CHANGES_REQUESTED** [review]
Please add functional tests first to check functionality

**Comment by SebastianGode** [comment]
@tischrei Integration Test added

File: plugins/modules/ces_quotas_info.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+EXAMPLES = '''
+# Query Alarm Quotas
+- opentelekomcloud.cloud.ces_quotas_info:
+'''
+
+from ansible_collections.opentelekomcloud.cloud.plugins.module_utils.otc import OTCModule
+
+
+class CesQuotasInfoModule(OTCModule):
+    argument_spec = dict(
```
Comment (anton-sidelnikov) [inline plugins/modules/ces_quotas_info.py:58]: you can write an one line argument_spec = dict()

File: plugins/modules/ces_alarms.py [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+    def run(self):
+        changed = False
+
+        if self.params['state'] == 'absent':
+            changed = False
+            al = self.conn.ces.find_alarm(self.params['alarm_name'])
+            self.conn.ces.delete_alarm(al)
+            changed = True
+
+        if self.params['state'] == 'present':
```
Comment (anton-sidelnikov) [inline plugins/modules/ces_alarms.py:228]: did you try disable and then enable alarm?

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: plugins/modules/ces_alarms.py (hunk 8 above)
Comment (SebastianGode) [reply plugins/modules/ces_alarms.py:228]: Oh God, thanks!
Somehow I missed this point.

Will do that asap

Comment (SebastianGode) [reply plugins/modules/ces_alarms.py:229]: added in 4439f9a5521a2bd4ea4e65fbd0c9812da26c0f29

File: plugins/modules/ces_quotas_info.py (hunk 7 above)
Comment (SebastianGode) [reply plugins/modules/ces_quotas_info.py]: changed in 13371b9467e97932b8ca370607a56fc46aa9dce3

**Review by anton-sidelnikov — APPROVED** [review]

**Comment by tischrei** [comment]
recheck

**Review by tischrei — DISMISSED** [review]

**Comment by SebastianGode** [comment]
recheck

**Comment by SebastianGode** [comment]
recheck

**Comment by SebastianGode** [comment]
recheck

**Comment by SebastianGode** [comment]
recheck

**Review by tischrei — APPROVED** [review]

**Pull request merged**